inngest-core/src/main/kotlin/com/inngest/Inngest.kt

```diff
@@ -1,18 +1,32 @@
 package com.inngest
 
+import com.beust.klaxon.Klaxon
+import java.io.IOException
+
 class Inngest(val appId: String) {
     // TODO - Fetch INNGEST_EVENT_KEY env variable on instantiation
 
-//    fun send(event: Event): EventAPIResponse {
-//        val requestBody = Klaxon().toJsonString(event)
-//        val client = HttpClient.newBuilder().build()
-//        val request =
-//            HttpRequest.newBuilder()
-//                .uri(URI.create("http://localhost:8288/e/"))
-//                .POST(HttpRequest.BodyPublishers.ofString(requestBody))
-//                .build()
-//        val response = client.send(request, HttpResponse.BodyHandlers.ofString())
-//        val body = Klaxon().parse<EventAPIResponse>(response)
-//        return body;
-//    }
+    val headers: RequestHeaders = Environment.inngestHeaders()
+    val httpClient = HttpClient(RequestConfig(headers))
+
+    inline fun <reified T> send(
+        url: String,
+        payload: Any,
+    ): T? {
+        val request = httpClient.build(url, payload)
+
+        return httpClient.send(request) lambda@{ response ->
```

Comment:
cool TIL https://kotlinlang.org/docs/returns.html#return-to-labels

Reply:
I wonder if it's possible to restructure the SDK to use these to manipulate control flow instead of exceptions. We can consider it later since things work as is